sorting/radix_sort2.cpp

```diff
@@ -0,0 +1,160 @@
+/**
+ * @file
+ * @brief Algorith of [Radix sort](https://en.wikipedia.org/wiki/Radix_sort)
+ * @author [Suyash Jaiswal](https://github.com/Suyashjaiswal)
+ * @details
+ * Sort the vector of integers using radix sort i.e. sorting digit by digit
+ using count sort as subroutine.
+ * Running time of radix is O(d*(n+b)) where b is the base for representing
+ numbers and d in the max digits in input inegers and n is number of integers.
+ * consider example for n = 5, array elements = 432,234,143,332,123
+ *sorting digit by digit
+ *sorting according to
+ *1) 1st digit place
+ *=> 432,332,143,123,234
+ *
+ *2) 2nd digit place
+ *=> 123,432,332,234,143
+ *
+ *3) 3rd digit place
+ *=> 123,143,234,332,432
+ *
+ *using count sort at each step, which is stable.
+ *stable => already sorted according to previous digits.
+*/
+
+// header files
+#include <iostream>  // for io operations
+#include <vector>    // for std::vector
+/**
+ * @namespace sorting
+ * @brief Sorting algorithms
+ */
+namespace sorting {
+/**
+ * @namespace radix_sort
+ * @brief Functions for [Radix sort](https://en.wikipedia.org/wiki/Radix_sort)
+ * algorithm
+ */
+namespace radix_sort {
+/**
+ * @brief Class to create a vector and sort it.
+ */
+class radix {
+ public:
+    int n;
+    std::vector<int> ar;
```

Comment:
`std::vector` it self keep track of it's size then why to waste `4 bytes` more just to keep track the same thing again.
seems redundant to me.